src/panels/ItemPanel.tsx

```diff
@@ -177,39 +177,37 @@ export default class ItemPanel extends React.Component<Props, State> {
       Object.keys(Shapes)
         .concat("unsorted")
         .forEach((type) => (result[node.scheme][type] = []));
-      if (node.open) {
-        node.elements
-          .sort((a, b) => this.sort(a, b))
-          .filter(
-            (id) =>
-              this.search(id) &&
-              (AppSettings.representation === Representation.FULL ||
-                (AppSettings.representation === Representation.COMPACT &&
-                  !(
-                    WorkspaceTerms[WorkspaceElements[id].iri].types.includes(
-                      parsePrefix("z-sgov-pojem", "typ-vztahu")
-                    ) ||
-                    WorkspaceTerms[WorkspaceElements[id].iri].types.includes(
-                      parsePrefix("z-sgov-pojem", "typ-vlastnosti")
-                    )
-                  )))
-          )
-          .forEach((elem) => {
-            const types = WorkspaceTerms[WorkspaceElements[elem].iri].types;
-            for (const key in Shapes) {
-              if (types.includes(key)) {
-                result[node.scheme][key].push(elem);
-                break;
-              }
+      node.elements
+        .sort((a, b) => this.sort(a, b))
+        .filter(
+          (id) =>
+            this.search(id) &&
```

Comment:
Whenever a condition spans over more than a single line, it is hard to read. It would be best to break similar complex conditions down to components and create a standalone function for the filter (even having it declared as a const variable before the sorting / filtering would be great).